bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/handler/HaperoThingHandler.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hapero.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.hapero.internal.config.HaperoThingConfig;
+import org.openhab.binding.hapero.internal.device.Device;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HaperoThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels and update the channel data from the
+ * data blocks of the input stream
+ *
+ * @author Daniel Walter - Initial contribution
+ */
+@NonNullByDefault
+abstract class HaperoThingHandler extends BaseThingHandler {
+
+    /** for debug data logging */
+    protected final Logger logger = LoggerFactory.getLogger(HaperoThingHandler.class);
+
+    /**
+     * Constructor
+     *
+     * @param thing
+     */
+    public HaperoThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+
+        super.bridgeStatusChanged(bridgeStatusInfo);
+    }
```

Comment:
What is the idea with this method override when calling the super implementation just after - which will then change status once again?

See:
https://github.com/openhab/openhab-core/blob/3b91692be142cc0b1293416655825d201cb75c5c/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/binding/BaseThingHandler.java#L643-L651

Reply:
I think this is how it is in the generated template for a binding. And also the documentation seems to request it: [https://www.openhab.org/docs/developer/bindings/](https://www.openhab.org/docs/developer/bindings/#bridge-status-changes)
Seems the documentation is outdated because the code in your link seems to already to do what the documentation requests manually to be done.

Reply:
So i assume i can just remove the override without breaking something, right ?

Reply:
Yes.